sdk/daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/LanguageVersion.scala

```diff
@@ -28,8 +28,14 @@ object LanguageVersion {
   def assertFromString(s: String): LanguageVersion = data.assertRight(fromString(s))
 
   implicit val Ordering: scala.Ordering[LanguageVersion] = {
+    case (LanguageVersion(Major.V1, leftMinor), LanguageVersion(Major.V1, rightMinor)) =>
+      Major.V2.minorVersionOrdering.compare(leftMinor, rightMinor)
+    case (LanguageVersion(Major.V1, _), _) =>
+      -1
     case (LanguageVersion(Major.V2, leftMinor), LanguageVersion(Major.V2, rightMinor)) =>
       Major.V2.minorVersionOrdering.compare(leftMinor, rightMinor)
+    case _ =>
+      1
```

Comment:
Shouldn't we rather throw when comparing two different major version? If versions become comparable across major versions then we could run into weird errors where the code assumes that 2.1 is backwards compatible with 1.dev for instance.

Reply:
I will say we should simply not compare language version, we should just compare minor version
I will change that to throw, but we should remove comparison of language version in another PR.

Reply:
Good point, this would be correct by construction then.